docs/baremetal-operator-on-kubernetes-cluster.md

```diff
@@ -36,8 +36,8 @@ inter-pod communication.
 
 Take the Cluster-IP Address of ironic pod running in default namespace
 and copy it in the config maps of Baremetal-operator by replace the default
-provisioning-IP in *ironic-deployment/default/ironic_bmo_configmap.env* as mentioned below.
-One may also have to change *provisioning-interface*
+provisioning-IP in *ironic-deployment/default/ironic_bmo_configmap.env* as
+mentioned below. One may also have to change *provisioning-interface*
 (by default it is PROVISIONING_INTERFACE=eth2) and its value will change
 as per the OS and hardware specs.
 
```

Comment:
@maelk Do you want me to remove this portion and keep only baremetal-operator/configmap update?

Reply:
I think it is still problematic, specifically the order creation of the configmap vs starting ironic and BMO. You need to have the proper config map file before deploying ironic. if you don't, just updating the configmap will not be taken into account by ironic because ironic would need to restart to take it into account.

Regarding your question about clusterIP, ironic should not be using clusterIP. It runs on host networking. You can have it use a Virtual IP with keepalived. This would be the proper way to deploy ironic.

Reply:
In my case, i had been reapplying the ironic pods after getting an node assigned to it. Mostly, ironic runs everytime in the same worker node, even after reapplying the yaml many times. So, just like static address it can be updated in baremetal-operator. However, one efficient way would be using "NodeSelector" in k8 yaml configuration under specs so that ironic runs only in the specified worker-node everytime. The same can be updated in the baremetal yaml. Do, you want me to update the doc as such?

Reply:
The idea with the VIP and deploying ironic with keepalived is to remove this constraint of having to always run on the same node. you don't need to set a node selector if you are using a VIP that will automatically be added to the node where Ironic runs. I would very much recommend to use this approach, it gives much more flexibility, for example if that node goes down, Ironic can be restarted somewhere else without issues.